aws-cloudformation-stack/src/main/java/software/amazon/cloudformation/stack/UpdateHandler.java

```
@@ -0,0 +1,91 @@
+        final Logger logger) {
+
+        this.logger = logger;
+        ResourceModel model = request.getDesiredResourceState();
+
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Update Stack", request.getStackId(), request.getClientRequestToken()));
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+            .then(progress ->
+                proxy.initiate("AWS-CloudFormation-Stack::UpdateTerminationProtection", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                    .translateToServiceRequest(Translator::translateToUpdateTerminationProtectionRequest)
+                    .makeServiceCall((awsRequest, client) -> {
+                        UpdateTerminationProtectionResponse awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::updateTerminationProtection);
+                        logger.log(String.format("TerminationProtection has successfully been updated for stack %s.", progress.getResourceModel().getStackId()));
+                        return awsResponse;
+                    })
+                    .handleError((awsRequest, exception, client, _model, context) -> handleError(awsRequest, exception, client, _model, context))
+                    .progress())
+            .then(progress ->
+                proxy.initiate("AWS-CloudFormation-Stack::Update", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                    .translateToServiceRequest(Translator::translateToUpdateRequest)
+                    .makeServiceCall((awsRequest, client) -> {
+                        UpdateStackResponse awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::updateStack);
+                        logger.log(String.format("%s has successfully been updated.", ResourceModel.TYPE_NAME));
+                        return awsResponse;
+                    })
```

we shoudl avoid the API call here too.  we should also ignore the exception where no updates are to be performed

---

We will keep the same as discussed. One example why we don't want to check because if TemplateURL doesn't change but the content changes.